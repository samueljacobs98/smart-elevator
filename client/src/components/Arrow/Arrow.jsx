import "./Arrow.scss";
import arrow from "../../assets/svgs/arrow.svg";

const Arrow = ({ isLeft, isActive }) => {
  return (
    <img
      className={`arrow ${isLeft ? "arrow--left" : ""} ${
        isActive ? "arrow--active" : ""
      }`}
      src={arrow}
      alt={`${isLeft ? "Left" : "Right"} arrow`}
    />
  );
};

export default Arrow;
