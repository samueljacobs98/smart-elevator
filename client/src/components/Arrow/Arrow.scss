@use "../../assets/scss/variables" as *;

.arrow {
  width: 1rem;
  margin: 0.5rem;
  filter: brightness(0.2);

  &--active {
    scale: 1.5;
    filter: brightness(1);
  }

  @media (min-width: $breakpoint-mobile) {
    width: 1.5rem;
  }

  @media (min-width: $breakpoint-tablet) {
    width: 2.5rem;
  }

  &--left {
    transform: rotate(180deg);
  }
}
