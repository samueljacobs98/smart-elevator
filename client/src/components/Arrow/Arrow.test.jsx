import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import arrowSvg from "../../assets/svgs/arrow.svg";
import Arrow from "./Arrow";

describe("Arrow", () => {
  const renderComponent = (isLeft = false, isActive = false) =>
    render(<Arrow isLeft={isLeft} isActive={isActive} />);

  it("renders without crashing", () => {
    renderComponent();

    expect(screen).toBeDefined();
  });

  it("applies correct class names for right and inactive state", () => {
    renderComponent(false, false);

    const arrow = screen.getByRole("img");

    expect(arrow).toHaveClass("arrow");
    expect(arrow).not.toHaveClass("arrow--left arrow--active");
  });

  it("applies correct class names for left and active state", () => {
    renderComponent(true, true);

    const arrow = screen.getByRole("img");

    expect(arrow).toHaveClass("arrow arrow--left arrow--active");
  });

  it("has the correct image source", () => {
    renderComponent();

    const arrow = screen.getByRole("img");

    expect(arrow).toHaveAttribute("src", arrowSvg);
  });

  it("has the correct alt text for a right arrow", () => {
    renderComponent();

    const arrow = screen.getByRole("img");

    expect(arrow).toHaveAttribute("alt", "Right arrow");
  });

  it("has the correct alt text for a left arrow", () => {
    renderComponent(true, true);

    const arrow = screen.getByRole("img");

    expect(arrow).toHaveAttribute("alt", "Left arrow");
  });

  it("should match snapshot", () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });
});
