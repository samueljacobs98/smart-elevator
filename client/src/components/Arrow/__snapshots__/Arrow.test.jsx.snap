// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow should match snapshot 1`] = `
<div>
  <img
    alt="Right arrow"
    class="arrow  "
    src="arrow.svg"
  />
</div>
`;
